Index: src/main/java/useCaseInteractor/Requests/RequestProcess.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/useCaseInteractor/Requests/RequestProcess.java b/src/main/java/useCaseInteractor/Requests/RequestProcess.java
--- a/src/main/java/useCaseInteractor/Requests/RequestProcess.java	
+++ b/src/main/java/useCaseInteractor/Requests/RequestProcess.java	
@@ -5,7 +5,7 @@
 import com.mongodb.MongoClient;
 import com.mongodb.MongoClientURI;
 import database.MongoDBAccess;
-import entity.Request;
+import entity.User.Request;
 import useCaseInteractor.User.userCollection;
 
 import java.net.UnknownHostException;
Index: src/main/java/useCaseInteractor/Search/launchSearch.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/useCaseInteractor/Search/launchSearch.java b/src/main/java/useCaseInteractor/Search/launchSearch.java
--- a/src/main/java/useCaseInteractor/Search/launchSearch.java	
+++ b/src/main/java/useCaseInteractor/Search/launchSearch.java	
@@ -4,7 +4,7 @@
 import com.mongodb.DBCollection;
 import com.mongodb.MongoClient;
 import com.mongodb.MongoClientURI;
-import Search;
+import entity.User.Search;
 import database.MongoDBAccess;
 
 import java.net.UnknownHostException;
